packages/headless/src/api/knowledge/createAnswerAPISlice.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  EventSourceMessage,
+  fetchEventSource,
+} from '@microsoft/fetch-event-source';
+import {ConfigurationSection} from '../../state/state-sections';
+import {createAbortController} from '../../utils/abort-controller-polyfill';
+import {baseKnowledgeAPISlice} from './baseKnowledgeAPISlice';
+
+type StateNeededByKnowledgeAPI = ConfigurationSection;
+
+interface ErrorPayload {
+  message?: string;
+  code?: number;
+}
+
+class FatalError extends Error {
+  constructor(public payload: ErrorPayload) {
+    super(payload.message);
+  }
+}
+
+const abortController = createAbortController();
+let retryCount = 0;
+
+const data = {
+  answerStyle: '',
+  contentFormat: '',
+  answer: '',
+  citations: [],
+  generated: false,
+  isStreaming: true,
+  isLoading: true,
+};
+
+type Data = typeof data;
```

Comment:
Do we really need the `variable` data ? Can't we just use an interface ?
```suggestion
interface GeneratedAnswerStream { 
[...]
}
```

Reply:
The variable is used as the initial state. following your other global variables comments, I just created an interface and passed the initial state object directly in the `queryFn` since its not used anywhere else